### PR TITLE
Add ramoops_ftrace and tracebuffer nodes to AbstractVM

### PR DESCRIPTION
**Pull request by jsvapiav**
* Add ramoops node with ftrace caps -- disabled
* Add sel4_tracebuffer node

jira: HYPR-432
Signed-off-by: Ivan Kuznetsov <jsvapiav@gmail.com>

### CODE REVIEW

File: components/VM_Arm/src/main.c [hunk 1]
```diff
@@ -852,6 +852,32 @@ static int generate_fdt(vm_t *vm, void *fdt_ori, void *gen_fdt, int buf_size, si
         err = fdt_generate_usb_node(gen_fdt);
         if (err)
             return -1;
+
+#define ADD_RAMOOPS_NODE  0
+#define ADD_SEL_LOGBUFFER 1
+
+#if ADD_SEL_LOGBUFFER
+	const unsigned long tracebuffer_base = 0x08800000;
```
Comment (hlyytine) [inline components/VM_Arm/src/main.c]: Maybe these are best defined in bcm2711 or rpi4-related header? Otherwise we will have interesting times ahead when we start to use some other platform.

File: components/VM_Arm/src/main.c [hunk 2]
```diff
@@ -852,6 +852,32 @@ static int generate_fdt(vm_t *vm, void *fdt_ori, void *gen_fdt, int buf_size, si
...
+
+	ZF_LOGE("Trying to add sel4 tracebuffer node: 0x%lx@0x%lx", tracebuffer_size, tracebuffer_base);
+	err = fdt_generate_sel4_tracebuffer_node(gen_fdt, tracebuffer_base, tracebuffer_size);
+	if (err) {
+		return -1;
+	}
+#endif // ADD_SEL_LOGBUFFER
+
+#if ADD_RAMOOPS_NODE
+	const unsigned long ramoops_base = 0x08000000;
```
Comment (hlyytine) [inline components/VM_Arm/src/main.c]: Ditto.

File: components/VM_Arm/src/main.c (hunk 1 above)
Comment (jsvapiav) [reply components/VM_Arm/src/main.c]: Those values are not related to bcm2711 or RPi4 at all. I was actually thinking that those values need to be defined and passed somewhere inside CAMKES configuration files. And this work still need to be done, but probably not in this commit.

Comment (jsvapiav) [reply components/VM_Arm/src/main.c]: But let me check if I will be able to add it to devices.camkes. Maybe it would be quite fast.

Comment (jsvapiav) [reply components/VM_Arm/src/main.c]: Removed. Now all this will be passed through `devices.camkes`

File: components/VM_Arm/src/main.c (hunk 2 above)
Comment (jsvapiav) [reply components/VM_Arm/src/main.c]: Removed. Now all this will be passed through `devices.camkes`

File: components/VM_Arm/src/main.c [hunk 3]
```diff
@@ -852,6 +857,23 @@ static int generate_fdt(vm_t *vm, void *fdt_ori, void *gen_fdt, int buf_size, si
         err = fdt_generate_usb_node(gen_fdt);
         if (err)
             return -1;
+
+        if (&tracebuffer_base && &tracebuffer_size && tracebuffer_base && tracebuffer_size) {
+
+        ZF_LOGE("Trying to add sel4 tracebuffer node: 0x%lx@0x%lx", tracebuffer_size, tracebuffer_base);
```
Comment (hlyytine) [inline components/VM_Arm/src/main.c]: Indentation seems a bit off?

File: components/VM_Arm/src/main.c [hunk 4]
```diff
@@ -852,6 +857,23 @@ static int generate_fdt(vm_t *vm, void *fdt_ori, void *gen_fdt, int buf_size, si
...
+        err = fdt_generate_sel4_tracebuffer_node(gen_fdt, tracebuffer_base, tracebuffer_size);
+             if (err) {
+                  return -1;
+             }
+        }
+
+        if (&ramoops_base && &ramoops_size && ramoops_base && ramoops_size) {
+        ZF_LOGE("Trying to add ramoops node: 0x%lx@0x%lx", ramoops_size, ramoops_base);
```
Comment (hlyytine) [inline components/VM_Arm/src/main.c]: Indentation problem here as well?

**Review by hlyytine — CHANGES_REQUESTED** [review]
Apart from those indentation issues, this is fine.

File: components/VM_Arm/src/main.c (hunk 3 above)
Comment (jsvapiav) [reply components/VM_Arm/src/main.c]: Fixed. Thanks.

File: components/VM_Arm/src/main.c (hunk 4 above)
Comment (jsvapiav) [reply components/VM_Arm/src/main.c]: Fixed. Thanks.

**Review by hlyytine — APPROVED** [review]
This is fine.

File: components/VM_Arm/src/fdt_manipulation.c [hunk 5]
```diff
@@ -47,59 +47,136 @@ int fdt_generate_memory_node(void *fdt, unsigned long base, size_t size)
...
+
 int fdt_generate_usb_node(void *fdt)
 {
     int root_offset = fdt_path_offset(fdt, "/scb/pcie@7d500000/pci@0,0");
     int address_cells = fdt_address_cells(fdt, root_offset);
     int size_cells = fdt_size_cells(fdt, root_offset);
 
-    if (root_offset <= 0)
-       ZF_LOGE("root_offset: %d", root_offset);
+    if (root_offset <= 0) {
```
Comment (hlyytine) [inline components/VM_Arm/src/fdt_manipulation.c:132]: Hmmm while these whitespace and similar style fixes are definitely good, can you put them into separate commit? That commit can be part of this PR, there's no problem. And the commit message could be something like "trivial: Fix whitespace".

**Review by hlyytine — CHANGES_REQUESTED** [review]

File: components/VM_Arm/src/fdt_manipulation.c (hunk 5 above)
Comment (jsvapiav) [reply components/VM_Arm/src/fdt_manipulation.c:132]: Done.

File: components/VM_Arm/src/main.c [hunk 6]
```diff
@@ -1193,6 +1216,7 @@ int main_continued(void)
     assert(!err);
     err = vm_register_notification_callback(&vm, handle_async_event, NULL);
     assert(!err);
+
```
Comment (hlyytine) [inline components/VM_Arm/src/main.c:1223]: Perhaps this line belongs to that style fix commit?

**Review by hlyytine — CHANGES_REQUESTED** [review]
